profiling/src/superluminal_impl.rs

```diff
@@ -9,6 +9,27 @@ macro_rules! scope {
     };
 }
 
+#[macro_export]
+macro_rules! function_scope {
+    () => {
+        let function_name = {
+            struct S;
+            let type_name = core::any::type_name::<S>();
+            &type_name[..type_name.len() - 3]
+        };
+        let _superluminal_guard = $crate::superluminal::SuperluminalGuard::new(function_name);
```

Comment:
It seems like the function_scope! implementations could just call into the scope! implementations

Reply:
If you already made this change (to have function_scope! call into scope!) it's fine, but after a second review, I think it's unnecessary indirection, I'd just keep it the way it is.